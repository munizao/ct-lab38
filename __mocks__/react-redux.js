jest.mock('react-redux', () => {
  return {
    __esModule: true,
    useDispatch: jest.fn(() => (action) => {}),
    useSelector: jest.fn((selector) => selector({ coffees: 0, snacks: 0, naps: 0, studies: 0 })),
  };
});
