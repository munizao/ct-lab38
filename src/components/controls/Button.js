import React from 'react';
import PropTypes from 'prop-types';

import { useDispatch, useSelector } from 'react-redux';
import { getGetCount } from '../../selectors/get-get-count';

const Button = ({ actionObj }) => {
  const { action, text, stateName } = actionObj;
  const dispatch = useDispatch();
  const count = useSelector(getGetCount(stateName));
  return (
    <button key={text} onClick={() => dispatch(action())}>
      {text}
      {!!count && `- ${count}`}
    </button>
  );
};

Button.propTypes = {
  actionObj: PropTypes.shape({
    action: PropTypes.func.isRequired,
    text: PropTypes.string.isRequired,
    stateName: PropTypes.string.isRequired
  })
};

export default Button;
