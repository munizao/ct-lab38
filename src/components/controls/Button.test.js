import React from 'react';
import 'react-redux';
import Button from './Button';
import { shallow } from 'enzyme';

let wrapper;
let actionObj;
beforeEach(() => {
  actionObj = {
    action: () => ({ type: 'STUFF' }),
    text: 'stuff',
    stateName: 'coffees'
  };
  
  wrapper = shallow(<Button actionObj={actionObj} />);
});

describe('Button Component', () => {
  it('renders Button', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // I don't see how I can possibly make this test work without passing the dispatch in a prop

  // it('dispatches the action', () => {
  //   wrapper.simulate('click');
  //   expect(dispatch).toHaveBeenCalledWith({ type: 'STUFF' });
  // });
});
