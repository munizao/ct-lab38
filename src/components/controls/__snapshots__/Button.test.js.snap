// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders Button 1`] = `
<button
  key="stuff"
  onClick={[Function]}
>
  stuff
</button>
`;
